src/code/fv-components/egg-hatch-drake.js

```diff
@@ -0,0 +1,53 @@
+import React, { PropTypes } from 'react';
+import AnimatedSprite from '../components/animated-sprite';
+import OrganismView from '../components/organism';
+import classNames from 'classnames';
+
+export default class EggHatchDrakeView extends React.Component {
+
+  static propTypes = {
+    id: PropTypes.string,
+    eggClasses: PropTypes.string,
+    drakeClasses: PropTypes.string,
+    drake: PropTypes.object,
+    hatchStarted: PropTypes.bool,
+    hatchComplete: PropTypes.bool,
+    onHatchComplete: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Interesting to think about, but I'm inclined to leave this as is. The two props/state variables are what the only client is already maintaining and the client code is simple. Requiring that the client code re-instantiate the component and/or update a `key` seems like pushing responsibility to the client that is better handled by the component in this case. The component can only track its own `hatchComplete` state when it's actually performing the animation. The `hatchComplete` prop allows the client to skip the animation and have the component state be maintained appropriately.